sonic_data_client/db_client.go

```diff
@@ -367,52 +368,92 @@ func ValToResp(val Value) (*gnmipb.SubscribeResponse, error) {
 	}
 }
 
-func GetTableKeySeparator(target string) (string, error) {
+func GetTableKeySeparator(target string, ns string) (string, error) {
 	_, ok := spb.Target_value[target]
 	if !ok {
 		log.V(1).Infof(" %v not a valid path target", target)
 		return "", fmt.Errorf("%v not a valid path target", target)
 	}
 
-	var separator string = sdcfg.GetDbSeparator(target)
+	var separator string = sdcfg.GetDbSeparator(target, ns)
 	return separator, nil
 }
+func GetRedisClientsForDb(target string) map[string]*redis.Client {
+	redis_client_map := make(map[string]*redis.Client)
+	if sdcfg.CheckDbMultiNamespace() {
+		ns_list, _ := sdcfg.GetDbNonDefaultNamespaces()
+		for _, ns := range ns_list {
+			redis_client_map[ns] = Target2RedisDb[ns][target]
+		}
+	} else {
+		ns := sdcfg.GetDbDefaultNamespace()
+		redis_client_map[ns] = Target2RedisDb[ns][target]
+	}
+	return redis_client_map
+}
+func IsTargetDb(target string) (string, bool, string, bool) {
```

Comment:
>IsTargetDb [](http://example.com/codeflow?start=5&length=10)

Could you explain the purpose of this function and each paramter and return value?

Reply:
This function get target present in GNMI Request and
returns: 1. DbName (string) 2. Is DbName valid (bool)
             3. DbNamespace (string) 4. Is DbNamespace present in Target

Reply:
What is "target present"?

Reply:
Hi @qiluo-msft For eg: In below gNMI request value associated with `-xpath_target`  is target which is `STATE_DB/asic3`.
So above function will return (`state_db`, `true`, `asic3`, `true`)


root@str2-xxxx-acs-3:/# gnmi_get -target_addr localhost:50051 -xpath SWITCH_CAPABILITY/switch -xpath_target STATE_DB/asic3 -insecure
== getRequest: